server/client_manager.py

```diff
@@ -276,55 +240,60 @@ def disemvowel_message(self, message):
 
     def __init__(self, server):
         self.clients = set()
-        self.cur_id = 0
         self.server = server
+        self.cur_id = [False] * self.server.config['playerlimit']
+        self.clients_list = []
 
     def new_client(self, transport):
-        c = self.Client(self.server, transport, self.cur_id)
+        cur_id = 0
+        for i in range(self.server.config['playerlimit']):
+                if not self.cur_id[i]:
+                    cur_id = i
+                    break
+        c = self.Client(self.server, transport, cur_id, self.server.get_ipid(transport.get_extra_info('peername')[0]))
         self.clients.add(c)
-        self.cur_id += 1
+        self.cur_id[cur_id] = True
         return c
 
     def remove_client(self, client):
+        self.cur_id[client.id] = False
         self.clients.remove(client)
-
-    def get_targets_by_ip(self, ip):
-        clients = []
-        for client in self.clients:
-            if client.get_ip() == ip:
-                clients.append(client)
-        return clients
-
-    def get_targets_by_hdid(self, hdid):
-        clients = []
-        for client in self.clients:
-            if client.get_hdid() == hdid:
-                clients.append(client)
-        return clients
-
-    def get_targets_by_ooc_name(self, name):
-        clients = []
-        for client in self.clients:
-            if client.name.lower() == name.lower():
-                clients.append(client)
-        return clients
-
-    def get_targets(self, client, target):
-        # check if it's IP but only if mod
-        if client.is_mod:
-            clients = self.get_targets_by_ip(target)
-            if clients:
-                return clients
-        # check if it's a character name in the same area
-        c = client.area.get_target_by_char_name(target)
-        if c:
-            return [c]
-        # check if it's an OOC name
-        ooc = self.get_targets_by_ooc_name(target)
-        if ooc:
-            return ooc
-        return None
-
+		
+    def get_targets(self, client, key, value, local):
+        #possible keys: ip, OOC, id, cname, ipid, hdid
```

Comment:
not really, use if-elif. the greatest advantage with enums is the absence of magic strings and numbers on the implementation side, as values are used symbolically